tests/system/test_access.py

```diff
@@ -16,44 +20,316 @@
+        cfg.update({"secret_token": "1234", "api_key_enabled": True, "enable_rum": True})
+        return cfg
+
+    def test_root(self):
+        """
+        Test authorization logic for root endpoint
+        """
+        url = self.root_url
+
+        for token in self.unauthorized_keys:
+            resp = requests.get(url, headers=headers(token))
+            assert resp.status_code == 200, "token: {}, status_code: {}".format(token, resp.status_code)
+            assert resp.content == '', "token: {}, response: {}".format(token, resp.content)
+
+        keys_one_privilege = [self.api_key_privilege_config,
+                              self.api_key_privilege_sourcemap, self.api_key_privilege_intake]
+        for token in self.authorized_keys+keys_one_privilege:
+            resp = requests.get(url, headers=headers(token))
+            assert resp.status_code == 200,  "token: {}, status_code: {}".format(token, resp.status_code)
+            assert resp.content != '',  "token: {}, response: {}".format(token, resp.content)
+            for token in ["build_date", "build_sha", "version"]:
+                assert token in resp.json(), "token: {}, response: {}".format(token, resp.content)
+
+    def test_backend_intake(self):
+        """
+        Test authorization logic for backend Intake endpoint
+        """
+        url = self.intake_url
+        events = self.get_event_payload()
+
+        for token in self.authorized_keys+[self.api_key_privilege_intake]:
+            resp = requests.post(url, data=events, headers=headers(token))
+            assert resp.status_code == 202,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+        for token in self.unauthorized_keys+[self.api_key_privilege_config, self.api_key_privilege_sourcemap]:
+            resp = requests.post(url, data=events, headers=headers(token))
+            assert resp.status_code == 401,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+    def test_rum_intake(self):
+        """
+        Test authorization logic for RUM Intake endpoint.
+        """
+        url = self.rum_intake_url
+        events = self.get_event_payload()
+
+        # Endpoint is not secured, all keys are expected to be allowed.
+        for token in self.authorized_keys + self.unauthorized_keys:
+            resp = requests.post(url, data=events, headers=headers(token))
+            assert resp.status_code != 401,  "token: {}, status_code: {}".format(token, resp.status_code)
+
+    def test_agent_config(self):
+        """
+        Test authorization logic for backend Agent Configuration endpoint
+        """
+        url = self.agent_config_url
+
+        for token in self.authorized_keys+[self.api_key_privilege_config]:
+            resp = requests.get(url, headers=headers(token, content_type="application/json"))
+            assert resp.status_code != 401,  "token: {}, status_code: {}".format(token, resp.status_code)
```

Comment:
Why is this `! 401`?  Will these be `404`?  Does it make sense to create configs and ensure they are returned?

Reply:
I did not intend to test the actual functionality of the endpoint here as it adds more coupling. Currently only the middleware checks for the API Key token, so I think the current approach is efficient. Once we also move a service name based check inside the handlers, it probably makes sense to also test handler functionality.